praas-app-ionic/src/pages/SignIn/index.tsx

```diff
@@ -10,53 +10,56 @@ import {
   IonInput,
   IonButton,
 } from '@ionic/react';
-import { useFormik } from 'formik';
 import { connect } from 'react-redux';
+import { withRouter, RouteComponentProps } from 'react-router';
+import { useForm } from 'react-hook-form';
 import Header from '../../components/Header';
 import { loginUser } from '../../store/user/login';
+import signinSchema from './schema';
 
 import './style.scss';
 
 type Props = {
-  loginUser: (data: any, formikActions: any) => void;
+  user: any;
+  loginUser: (data: any) => void;
 };
 
-const SignIn: React.FC<Props> = ({ loginUser }) => {
-  const formik = useFormik({
-    initialValues: { email: '', password: '' },
-    onSubmit: (values, actions) => {
-      loginUser({ user: values }, actions);
-    },
+const SignIn: React.FC<Props & RouteComponentProps> = ({
```

Comment:
@vakareeralil Not knowing TS in depth, dumb question. Is there a performance impact between the two alternatives? In general the React world seems to be moving to more of function components than class components. Not sure how that applies in TS.

Reply:
The impact of these on the final transpiled code would be the least is what I can say. Different parts of the bundling handle it differently. If babel does this job, the typescript related annotations are just stripped off and then transpiled. If it passes through another typescript compiler layer, it could be acting differently. It depends on how it is configured in ionic. But in either ways, the code generated by these should be definitely optimal which wouldnt have any type references and is purely JS. All the types though could be emitted to a separate file which could be helpful, but depends on our usecase. And this again is not causing any impact on the build nor performance. This is from what information I have. Though I wouldnt mind diving deeper if you have any specific questions related to this and wants me to do it.